### PR TITLE
Spacing

### PR DESCRIPTION
**Pull request by inikolova**
Related to https://github.com/telerik/kendo-themes-private/issues/176

Design issue https://github.com/telerik/web-components-ux/issues/944

### CODE REVIEW

**Review by JoomFX — APPROVED** [review]
LGTM

I guess the `$kendo-padding-x` and `$kendo-padding-y` in Default, Material and Classic does not use the Spacing system (but hardcoded values instead) on purpose, right?

File: packages/fluent/scss/menu/_variables.scss [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 /// Width of the menu border.
 /// @group menu
-$kendo-menu-border-width: map.get( $kendo-spacing, 0 ) !default;
+$kendo-menu-border-width: k-spacing(.0) !default;
```
Comment (epetrow) [inline packages/fluent/scss/menu/_variables.scss:8]: There is a typo here, should be `k-spacing(0)`

File: packages/fluent/scss/menu/_variables.scss [hunk 2]
```diff
@@ -18,26 +18,26 @@ $kendo-menu-line-height: var( --kendo-line-height, normal ) !default;
...
 /// @group menu
-$kendo-menu-item-padding-x: map.get( $kendo-spacing, 2 ) !default;
-$kendo-menu-item-padding-y: map.get( $kendo-spacing, 3 ) !default;
+$kendo-menu-item-padding-x: k-spacing(2) !default;
+$kendo-menu-item-padding-y: k-spacing(3) !default;
 
 /// Spacing between menu root items.
 /// @group menu
-$kendo-menu-item-spacing: map.get( $kendo-spacing, 0 ) !default;
+$kendo-menu-item-spacing: k-spacing(.0) !default;
```
Comment (epetrow) [inline packages/fluent/scss/menu/_variables.scss:26]: Same here, change to `k-spacing(0)`

File: packages/fluent/scss/orgchart/_variables.scss [hunk 3]
```diff
@@ -136,7 +136,7 @@ $kendo-orgchart-card-body-border-color: transparent !default;
 
 /// The horizontal margin of the OrgChart Card body title wrap.
 /// @group orgchart
-$kendo-orgchart-card-body-vbox-margin-x: $kendo-orgchart-spacer - map.get( $kendo-spacing, 2 ) !default;
+$kendo-orgchart-card-body-vbox-margin-x: $kendo-orgchart-spacer - k-spacing(2) !default;
```
Comment (epetrow) [inline packages/fluent/scss/orgchart/_variables.scss:139]: This does not seem correct, please check

File: packages/fluent/scss/tabstrip/_layout.scss [hunk 4]
```diff
@@ -1,5 +1,6 @@
 @use "../core/" as *;
 @use "./_variables.scss" as *;
+@use "../core/spacing" as *;
```
Comment (epetrow) [inline packages/fluent/scss/tabstrip/_layout.scss:3]: Is this import needed?

File: packages/material/scss/breadcrumb/_variables.scss [hunk 5]
```diff
@@ -96,13 +96,13 @@ $kendo-breadcrumb-link-border-radius: $kendo-border-radius-md !default;
 $kendo-breadcrumb-icon-link-padding-y: k-map-get( $kendo-spacing, 2.5 ) !default;
```
Comment (epetrow) [inline packages/material/scss/breadcrumb/_variables.scss:96]: Use `k-spacing()`?

File: packages/material/scss/core/spacing/index.import.scss [hunk 6]
```diff
@@ -0,0 +1,14 @@
+@import "@progress/kendo-theme-core/scss/spacing/index.import.scss";
+
+$kendo-icon-spacing: k-spacing(2) !default;
+$kendo-icon-padding: k-spacing(1) !default;
+
+// Metrics
+$kendo-padding-x: 16px !default;
```
Comment (epetrow) [inline packages/material/scss/core/spacing/index.import.scss:7]: Shouldn't we use values from `$kendo-spacing` for padding x and y here?

File: packages/material/scss/draggable/_variables.scss [hunk 7]
```diff
@@ -1,6 +1,6 @@
 // Drag hint
-$kendo-drag-hint-padding-x: $kendo-padding-md-x !default;
-$kendo-drag-hint-padding-y: k-math-div( $kendo-padding-md-x, 2 ) !default;
+$kendo-drag-hint-padding-x: #{$kendo-padding-md-x} !default;
```
Comment (epetrow) [inline packages/material/scss/draggable/_variables.scss:2]: Do we need to interpolate here?

File: packages/material/scss/list/_variables.scss [hunk 8]
```diff
@@ -85,15 +85,15 @@ $kendo-list-lg-item-line-height: null !default;
...
+$kendo-list-lg-group-item-padding-x: k-spacing(4) !default;
 
 /// The vertical padding of the List group items, when no size is set.
 /// @group list
 $kendo-list-group-item-padding-y: null !default;
-$kendo-list-sm-group-item-padding-y: k-map-get( $kendo-spacing, 1.5 ) !default;
-$kendo-list-md-group-item-padding-y: k-map-get( $kendo-spacing, 2 ) !default;
+$kendo-list-sm-group-item-padding-y: k-spacing(1.5) !default;
+$kendo-list-md-group-item-padding-y: k-spacing(2) !default;
 $kendo-list-lg-group-item-padding-y: k-map-get( $kendo-spacing, 2.5 ) !default;
```
Comment (epetrow) [inline packages/material/scss/list/_variables.scss:97]: Replace with `k-spacing(2.5)` here?

File: packages/material/scss/pager/_variables.scss [hunk 9]
```diff
@@ -5,13 +5,13 @@
...
 /// @group pager
-$kendo-pager-sm-padding-x: k-map-get( $kendo-spacing, 1 ) !default;
+$kendo-pager-sm-padding-x: k-spacing(1) !default;
 /// The horizontal padding of the medium Pager.
 /// @group pager
-$kendo-pager-md-padding-x: k-map-get( $kendo-spacing, 2 ) !default;
+$kendo-pager-md-padding-x: k-spacing(2) !default;
 /// The horizontal padding of the large Pager.
 /// @group pager
 $kendo-pager-lg-padding-x: k-map-get( $kendo-spacing, 2.5 ) !default;
```
Comment (epetrow) [inline packages/material/scss/pager/_variables.scss:17]: Replace with `k-spacing()`, same for `$kendo-pager-lg-item-group-spacing`

File: packages/material/scss/rating/_variables.scss [hunk 10]
```diff
@@ -12,14 +12,14 @@ $kendo-rating-line-height: $kendo-line-height-md !default;
...
+$kendo-rating-container-margin-x: calc( #{$kendo-padding-sm-x} / 2 ) !default;
 
 /// The horizontal padding of the Rating item.
 /// @group rating
-$kendo-rating-item-padding-x: k-math-div( $kendo-padding-sm-x, 2 ) !default;
+$kendo-rating-item-padding-x: calc( #{$kendo-padding-sm-x} / 2 ) !default;
 /// The vertical padding of the Rating item.
 /// @group rating
-$kendo-rating-item-padding-y: $kendo-padding-md-y !default;
+$kendo-rating-item-padding-y: #{$kendo-padding-md-y} !default;
```
Comment (epetrow) [inline packages/material/scss/rating/_variables.scss:22]: I don't think interpolation is needed here, please check

File: packages/material/scss/toolbar/_variables.scss [hunk 11]
```diff
@@ -5,10 +5,10 @@
...
 /// @group toolbar
-$kendo-toolbar-sm-padding-x: k-map-get( $kendo-spacing, 1 ) !default;
+$kendo-toolbar-sm-padding-x: k-spacing(1) !default;
 /// The horizontal padding of the medium Toolbar.
 /// @group toolbar
-$kendo-toolbar-md-padding-x: k-map-get( $kendo-spacing, 2 ) !default;
+$kendo-toolbar-md-padding-x: k-spacing(2) !default;
 /// The horizontal padding of the large Toolbar.
 /// @group toolbar
 $kendo-toolbar-lg-padding-x: k-map-get( $kendo-spacing, 2.5 ) !default;
```
Comment (epetrow) [inline packages/material/scss/toolbar/_variables.scss:14]: Replace this with `k-spacing()` as well for `$kendo-toolbar-lg-spacing`

**Review by epetrow — CHANGES_REQUESTED** [review]

File: packages/material/scss/table/_variables.scss [hunk 12]
```diff
@@ -30,12 +30,12 @@ $kendo-table-cell-padding-x: null !default;
 $kendo-table-cell-padding-y: null !default;
 
 $kendo-table-sm-cell-padding-x: k-map-get( $kendo-spacing, 2.5 ) !default;
```
Comment (zhpenkov) [inline packages/material/scss/table/_variables.scss:32]: There are still some values left, not using `k-spacing( n )` function.
Please search e.g `$kendo-spacing,` globally at your local repo and replace them where necessary.

**Filter:**
image

Comment (inikolova) [reply packages/material/scss/table/_variables.scss]: Not all of them should be replaced, however I replaced where needed :)

**Review by zhpenkov — COMMENTED** [review]
I could find some more common values that can be replaced with `k-spacing()`, using the following variable name prefixes:

- `-padding:`

- `-margin:`

- `-gap:`

- `-spacing:`

- `-spacing-x:`

- `-spacing-y:`

Can you please double check them using `\w+{prefix}\s*(-?\d*\.?\d+(?:px|em|rem|%)?)\s*!default;` regex in global search and replace them with `k-spacing()` where/if needed.

**Comment by inikolova** [comment]
> I could find some more common values that can be replaced with `k-spacing()`, using the following variable name suffixes:
> 
> * `-padding:`
> * `-margin:`
> * `-gap:`
> * `-spacing:`
> * `-spacing-x:`
> * `-spacing-y:`
> 
> Can you please double check them using `\w+{suffix}\s*(-?\d*\.?\d+(?:px|em|rem|%)?)\s*!default;` regex in global search and replace them with `k-spacing()` where/if needed.

I replaced all margins, paddings, gaps with common values. As for the `-spacing-` variables, I would prefer to replace them on another iteration

**Review by zhpenkov — APPROVED** [review]

**Review by epetrow — APPROVED** [review]

**Pull request merged**